src/Core/StorageManager.php

```diff
@@ -73,65 +71,111 @@ class StorageManager
 	 * @param LoggerInterface $logger
 	 * @param L10n            $l10n
 	 */
-	public function __construct(Database $dba, IConfig $config, LoggerInterface $logger, L10n $l10n, IHTTPRequest $httpRequest)
+	public function __construct(Database $dba, IConfig $config, LoggerInterface $logger, L10n $l10n)
 	{
 		$this->dba         = $dba;
 		$this->config      = $config;
 		$this->logger      = $logger;
 		$this->l10n        = $l10n;
-		$this->httpRequest = $httpRequest;
 		$this->backends = $config->get('storage', 'backends', self::DEFAULT_BACKENDS);
 
 		$currentName = $this->config->get('storage', 'name', '');
 
 		// you can only use user backends as a "default" backend, so the second parameter is true
-		$this->currentBackend = $this->getByName($currentName, true);
+		$this->currentBackend = $this->getSelectableStorageByName($currentName);
 	}
 
 	/**
 	 * Return current storage backend class
 	 *
-	 * @return Storage\IStorage|null
+	 * @return Storage\ISelectableStorage|null
 	 */
 	public function getBackend()
 	{
 		return $this->currentBackend;
 	}
 
+	/**
+	 * Returns a selectable storage backend class by registered name
+	 *
+	 * @param string|null $name            Backend name
+	 *
+	 * @return Storage\ISelectableStorage|null null if no backend registered at $name
+	 *
+	 * @throws \Friendica\Network\HTTPException\InternalServerErrorException
+	 */
+	public function getSelectableStorageByName(string $name = null)
```

Comment:
Shouldn't this method throw an exception when the backend isn't valid?

Reply:
Done

Reply:
I reverted this change, because there are a lot of places (mainly inside the legacy api), where an "invalid" backend results in using the legacy storage method.

Reply:
Can't this be done in a try-catch block instead of checking the returned value of the method?

Reply:
You're right .. but I'd need to adapt all calls, I tried to avoid it :D

Reply:
When a function main use is this:
```php
$return = function($param);
if ($return) {
   ...
} else {
   // Awkward error handling
}
```
Then it really should be
```php
try {
    $return = function($param);
    ...
} catch (Throwable $e) {
    // Proper error handling
}
```
The structures are similar and easy to transform from one to the other, it just is more explicit in the latter case.

Reply:
of course, you're right, but I need to take into account that in case of the "legacy" backend, there's no "current backend" set, so it would throw this exception all the time when calling `DI::storage()` ..